mlir/lib/Dialect/Presburger/Set.cpp

```diff
@@ -0,0 +1,396 @@
+#include "mlir/Dialect/Presburger/Set.h"
+#include "mlir/Analysis/Presburger/Simplex.h"
+
+// TODO should we change this to a storage type?
+using namespace mlir;
+using namespace mlir::presburger;
+
+unsigned PresburgerSet::getNumBasicSets() const {
+  return flatAffineConstraints.size();
+}
+
+unsigned PresburgerSet::getNumDims() const { return nDim; }
+
+unsigned PresburgerSet::getNumSyms() const { return nSym; }
+
+bool PresburgerSet::isMarkedEmpty() const { return markedEmpty; }
+
+bool PresburgerSet::isUniverse() const {
+  return flatAffineConstraints.size() == 0 && !markedEmpty;
+}
+
+const SmallVector<FlatAffineConstraints, 4> &
+PresburgerSet::getFlatAffineConstraints() const {
+  return flatAffineConstraints;
+}
+
+void PresburgerSet::addFlatAffineConstraints(FlatAffineConstraints cs) {
+  assert(cs.getNumDimIds() == nDim &&
+         "Cannot add FlatAffineConstraints having different dimensionality");
+
+  if (cs.isEmptyByGCDTest())
+    return;
+
+  markedEmpty = false;
+  flatAffineConstraints.push_back(cs);
+}
+
+void PresburgerSet::unionSet(const PresburgerSet &set) {
+  assert(set.getNumDims() == nDim &&
+         "Cannot union Presburger sets having different dimensionality");
+  assert(set.getNumSyms() + nSym == 0 &&
```

Comment:
This works if we ignore the naming of the parameters/symbols. If we are only considering the position of the symbol we can do this nicely, otherwise we need more functionality to do so.

Reply:
I'm doing it as you suggested for now.